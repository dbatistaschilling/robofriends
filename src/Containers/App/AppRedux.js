import React, { Component } from "react";
import "./App.css";
import CardList from "../../Components/CardList";
import SearchBox from "../../Components/SearchBox";
import Scroll from "../../Components/Scroll";
import ErrorBoundry from "../../Components/ErrorBoundry";
import Header from '../../Components/Header';

export class AppRedux extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    })
  }

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading...</h1> :
            <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

export default AppRedux;
