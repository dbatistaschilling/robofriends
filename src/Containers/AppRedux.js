import React, { useEffect } from "react";
import { connect } from 'react-redux';
import "./App.css";
import CardList from "../Components/CardList";
import SearchBox from "../Components/SearchBox";
import Scroll from "../Components/Scroll";
import ErrorBoundry from "../Components/ErrorBoundry";
import Header from '../Components/Header';

import { setSearchField, requestRobots } from '../actions';

const mapStateToProps = state => {
  return {
    searchField: state.searchRobots.searchField,
    robots: state.requestRobots.robots,
    isPending: state.requestRobots.isPending,
    error: state.requestRobots.error
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onSearchChange: (event) => dispatch(setSearchField(event.target.value)),
    onRequestRobots: () => dispatch(requestRobots())
  }
}

function App(props) {

  const { searchField, onSearchChange, onRequestRobots, robots, isPending } = props;

  useEffect(() => {
    onRequestRobots();
  }, [onRequestRobots])


  const filteredRobots = robots.filter((robot) => {
    return robot.name
      .toLowerCase()
      .includes(searchField.toLowerCase());
  });

  return isPending ?
    <h1 className="tc f1">Loading...</h1>
  :
    (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundry>
            <CardList robots={filteredRobots} />
          </ErrorBoundry>
        </Scroll>
      </div>
    )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps)(App);
