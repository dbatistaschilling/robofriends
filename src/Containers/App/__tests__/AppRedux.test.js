import React from 'react';
import { shallow } from 'enzyme';
import AppRedux from '../AppRedux';

let wrappedAppRedux;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: true
  }
  wrappedAppRedux = shallow(< AppRedux {...mockProps} />)
})

describe('AppRedux Tests', () => {
	it('Renders the App Redux page with no robots without crashing', () => {
    expect(wrappedAppRedux).toMatchSnapshot();
	})

	it('Filters robots correctly', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [{
        id: 3,
        name: 'john',
        email: 'john@gmail.com'
      }],
      searchField: 'john',
      isPending: false
    }
    const wrappedAppRedux2 = shallow(< AppRedux {...mockProps2} />)
    expect(wrappedAppRedux2).toMatchSnapshot();

		expect(wrappedAppRedux.instance().filterRobots()).toEqual([]);
		expect(wrappedAppRedux2.instance().filterRobots()).toEqual([{
      id: 3,
      name: 'john',
      email: 'john@gmail.com'
    }]);
  })
})