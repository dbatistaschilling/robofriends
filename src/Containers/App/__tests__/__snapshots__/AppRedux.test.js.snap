// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRedux Tests Filters robots correctly 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={
          Array [
            Object {
              "email": "john@gmail.com",
              "id": 3,
              "name": "john",
            },
          ]
        }
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`AppRedux Tests Renders the App Redux page with no robots without crashing 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <SearchBox />
  <Scroll>
    <h1>
      Loading...
    </h1>
  </Scroll>
</div>
`;
